doc/UserAgents.md

```diff
@@ -0,0 +1,102 @@
+
+> The User-Agent request-header field contains information about the
+user agent originating the request. This is for statistical purposes,
+the tracing of protocol violations, and automated recognition of user
+agents for the sake of tailoring responses to avoid particular user
+agent limitations. User agents SHOULD include this field with
+requests. The field can contain multiple product tokens (section 3.8)
+and comments identifying the agent and any subproducts which form a
+significant part of the user agent. By convention, the product tokens
+are listed in order of their significance for identifying the
+application. 
+```
+    User-Agent: CERN-LineMode/2.15 libwww/2.17b3
+```
+
+Chrome User-Agent String:
+```
+User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.79 Safari/537.
+```
+Edge User-Agent String:
+```
+User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; ServiceUI 13) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.140 Safari/537.36 Edge/17.17134
+```
+Internet Explorer User-Agent String:
+```
+User-Agent: Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; rv:11.0) like Gecko
+```
+
+Chrome Extensions modify the User-Agent string through code:
+```js
+chrome.webRequest.onBeforeSendHeaders.addListener(
+  function(details) {
+    for (var i = 0; i < details.requestHeaders.length; ++i) {
+      if (details.requestHeaders[i].name === 'User-Agent') {
+        details.requestHeaders[i].value = details.requestHeaders[i].value + ' OurUAToken/1.0';
+      }
+      break;
+    }
+    return { requestHeaders: details.requestHeaders };
+  });
+```
+
+
+## Current User Agent String
+V4 BotBuilder SDK does the following:
+```
+"user-agent": "Microsoft-BotFramework/3.1 (BotBuilder .Net/4.0.0.0)"
+```
+
+## Proposed Bot User Agent 
+Add the Following:
+1. SDK Version. The Bot Framework SDK Version MUST be added to the user-agent string. 
+2. SDK Host. The SDK Host, .Net Core, JVM, Node version, MUST be added to the user-agent string. 
+3. Operating System. The OS that hosts the Bot SHOULD be added to the user-agent string. 
+4. Azure Bot Service Version (if applicable)
+5. Active Middleware Components. A RFC 2616 compatable list of Middleware SHOULD be added. The format is:
+    ```
+    (Middleware1/version; Middleware2/version; Middleware3/version)
+    ```
+6. Active Storage Components. The storage providers being used SHOULD be added to the user-agent string. 
+    ```
+    (CosmosDBStorageProvider/version; AzureTables/version;)
+    ```
+7. LUIS. The usage of LUIS SHOULD be added to the user-agent string. 
+    ```
+    (LUIS/APIversion)
+    ```
+8. QnA. The usage of QnA Maker SHOULD be added to the user-agent string. 
+    ```
+    (QnAMaker/APIVersion)
+    ```
+9. Classic. Usage of the "Classic / Compatability" layers SHOULD be added to the user-agent string.
+    ```
+    classic/version
+    ```
+
```

Comment:
@cleemullins  can you add a 'generic' service example? In the future, we might add services such as Personality and specific middleware for Dispatch, or any other generic 'future' service. The pattern is very clear, just for completeness?